tests/gtests/graph/unit/utils/test_pattern_matcher_cpu.cpp

```diff
@@ -1108,6 +1108,46 @@ TEST(test_utils_pattern_matcher, OptionalWithLargerPort) {
     ASSERT_EQ(fusion_ops.size(), 3U);
 }
 
+TEST(test_utils_pattern_matcher, CommutativeSelect) {
+    auto graphp = std::make_shared<pb_graph_t>();
+    // Pattern that captures
+    // MatMul  Add
+    //   \     /
+    //    Select
+    auto pmatmul = graphp->append_op(MatMul);
+    auto aadd = graphp->append_op(Add);
+    auto pselect = graphp->append_op(
+            Select, {in_edge(1, pmatmul, 0), in_edge(2, aadd, 0)});
+    UNUSED(pselect);
+
+    graph_t agraph;
+    op_t matmul {0, MatMul, "matmul"};
+    op_t add {1, Add, "Add"};
+    op_t select {2, Select, "select"};
+
+    std::vector<logical_tensor_t> lt_vec = create_logical_tensors(8);
+    lt_vec[6].data_type = data_type::boolean;
+    matmul.add_input(lt_vec[0]);
+    matmul.add_input(lt_vec[1]);
+    matmul.add_output(lt_vec[2]);
+    add.add_input(lt_vec[3]);
+    add.add_input(lt_vec[4]);
+    add.add_output(lt_vec[5]);
+    select.add_input(lt_vec[6]);
+    select.add_input(lt_vec[5]);
+    select.add_input(lt_vec[2]);
+    select.add_output(lt_vec[7]);
+
+    ASSERT_EQ(agraph.add_op(&matmul), status::success);
+    ASSERT_EQ(agraph.add_op(&add), status::success);
+    ASSERT_EQ(agraph.add_op(&select), status::success);
+    agraph.finalize();
+
+    std::vector<op_t *> fusion_ops;
+    EXPECT_TRUE(match_pattern(agraph.get_ops()[0].get(), graphp, fusion_ops));
```

Comment:
And here we need to match from the select op, to verify our assumption.

Reply:
I don't understand "match from the select op". Is there anything to be modified?

Reply:
It seems impossible to declare `select` first and `edges` at last because edge related functions of `pb_graph_t` are protected.